tools/export_program_info/export_program_info.cpp

```diff
@@ -0,0 +1,162 @@
+    &_ebpf_sock_ops_program_info,
+    &_ebpf_xdp_program_info,
+    &_sample_ebpf_extension_program_info};
+
+ebpf_program_section_info_t _sample_ext_section_info[] = {
+    {L"sample_ext", &EBPF_PROGRAM_TYPE_SAMPLE, &EBPF_ATTACH_TYPE_SAMPLE, BPF_PROG_TYPE_SAMPLE, BPF_ATTACH_TYPE_SAMPLE}};
+
+static std::vector<ebpf_program_section_info_with_count_t> _section_information = {
+    {&_ebpf_bind_section_info[0], SIZE_OF_ARRAY(_ebpf_bind_section_info)},
+    {&_ebpf_xdp_section_info[0], SIZE_OF_ARRAY(_ebpf_xdp_section_info)},
+    {&_ebpf_sock_addr_section_info[0], SIZE_OF_ARRAY(_ebpf_sock_addr_section_info)},
+    {&_ebpf_sock_ops_section_info[0], SIZE_OF_ARRAY(_ebpf_sock_ops_section_info)},
+    {&_sample_ext_section_info[0], SIZE_OF_ARRAY(_sample_ext_section_info)},
+};
+
+uint32_t
+export_all_program_information()
+{
+    uint32_t status = ERROR_SUCCESS;
+    size_t array_size = sizeof(program_information_array) / sizeof(program_information_array[0]);
+    for (uint32_t i = 0; i < array_size; i++) {
+        auto program_type = program_information_array[i];
+        switch (program_type->program_type_descriptor.bpf_prog_type) {
+        case BPF_PROG_TYPE_BIND:
+            program_type->program_type_descriptor.program_type = EBPF_PROGRAM_TYPE_BIND;
+            break;
+        case BPF_PROG_TYPE_XDP:
+            program_type->program_type_descriptor.program_type = EBPF_PROGRAM_TYPE_XDP;
+            break;
+        case BPF_PROG_TYPE_CGROUP_SOCK_ADDR:
+            program_type->program_type_descriptor.program_type = EBPF_PROGRAM_TYPE_CGROUP_SOCK_ADDR;
+            break;
+        case BPF_PROG_TYPE_SOCK_OPS:
+            program_type->program_type_descriptor.program_type = EBPF_PROGRAM_TYPE_SOCK_OPS;
+            break;
+        case BPF_PROG_TYPE_SAMPLE:
+            program_type->program_type_descriptor.program_type = EBPF_PROGRAM_TYPE_SAMPLE;
+            break;
+        };
+
+        status = ebpf_store_update_program_information(program_information_array[i], 1);
+        if (status != ERROR_SUCCESS) {
+            break;
+        }
+    }
+
+    return status;
+}
+
+uint32_t
+export_all_section_information()
+{
+    uint32_t status = ERROR_SUCCESS;
+    for (const auto& section : _section_information) {
+        status = ebpf_store_update_section_information(section.section_info, (uint32_t)section.section_info_count);
+        if (status != ERROR_SUCCESS) {
+            break;
+        }
+    }
+
+    return status;
+}
+
+int
+export_global_helper_information()
+{
+    return ebpf_store_update_global_helper_information(
+        ebpf_core_helper_function_prototype, ebpf_core_helper_functions_count);
+}
+
+static uint32_t
+_clear_ebpf_store(ebpf_registry_key_t root_key_path)
+{
+    ebpf_registry_key_t root_handle = {0};
+    ebpf_registry_key_t provider_handle = {0};
+    uint32_t status;
+
+    // Open root registry key.
+    status = open_registry_key(root_key_path, EBPF_ROOT_RELATIVE_PATH, REG_CREATE_FLAGS, &root_handle);
```

Comment:
Can you use ebpf_store_helper.h?

Reply:
My comments about using ebpf_store_helper.h and hiding registry functions would be ok to defer to a later PR, they are not blocking.

Reply:
Filed #1259 to track these changes.